internal/olm/client/client.go

```diff
@@ -214,7 +216,67 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 		return curPhase == olmapiv1alpha1.CSVPhaseSucceeded, nil
 	}
 
-	return wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	err := wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	if err != nil && errors.Is(err, context.DeadlineExceeded) {
+		if depCheckErr := c.printDeploymentErrors(ctx, key, csv); depCheckErr != nil {
+			return fmt.Errorf("failed to run operator: %v %v", err, depCheckErr)
+		}
+	}
+	return err
+}
+
+// checkDeploymentErrors function loops through deployment specs of a given CSV, and prints reason
+// in case of failures, based on deployment condition.
+func (c Client) printDeploymentErrors(ctx context.Context, key types.NamespacedName, csv olmapiv1alpha1.ClusterServiceVersion) error {
```

Comment:
Maybe I missed something. This was `checkDeploymentErrors` before. It feels weird to me for a "client" to be printing out messages to the screen. I would not expect a kube client to be printing to the screen. I'd expect to get back errors and I can decide what to do with those errors. Not sure where the change to print came in, sorry if I'm late to the discussion.

Reply:
Looking at the other part of the `Client` looks like we already print to the screen so these changes are not out of the ordinary for this structure.